app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java

```diff
@@ -66,7 +69,18 @@ public void initiateCustomGalleryPickWithPermission(final Activity activity) {
 
```

Comment:
Yes, it is executed on pressing the custom selector. I suggest @sivaraam run the app with these changes & observe if the case is still the same. If so we have two options:

1. Request permissions on clicking the **+** floating action button in ContributionsListFragment, so it is granted before the user could use any of the image pickers

2. Copy the ACCESS_MEDIA_LOCATION permission request code block to the function that gets executed when the regular picker is pressed, ie: initiateGalleryPick() at line 60 in ContributionController.java

I guess the second option is the most appropriate as there is no need to obtain WRITE_EXTERNAL_STORAGE permission if the user never uses the custom picker.

Reply:
I tried to to share an image (which has location information) to the app while it does not have the location permission. When clicking on the map icon in the upload wizard, the app shows me the location drop pin shows an approx. location that is a bit far from the actual location. The actual location is not indicated at all.

When I repeat the same when the app has been granted the location permission, the location drop pin is situated a bit far away. Though, the actual location is at least indicated correctly.

PS: This seems to be the case for me when trying to upload via custom selector too (I'm not shown any location request dialog).

Reply:
Hi @sivaraam, this is happening with me as well. Is this not because the EXIF location may be different from user's current location (in case the user uploads the picture later from somewhere else)? The location drop pin would display the EXIF location and the actual location is indicated by the circular mark. As far as the request dialog is concerned, adding it in the beginning (as I'd demonstrated [here](https://github.com/commons-app/apps-android-commons/pull/5205#discussion_r1167472927)) or at some other appropriate stage will indicate the actual location too. The circular mark indicating the actual location will then not be visible only in case the user denies to provide location access.